### ISSUE TITLE
ScrapliTimeout raised by edit_config when config reaches a certain length.

### ISSUE THREAD

**Issue by mobious01** [scrapli/scrapli_netconf#46]
**Describe the bug**
A ScrapliTimeout is raised when the length of the config used with edit_config reaches a certain length.

**To Reproduce**
1. Your script
```python
from lxml.etree import Element, tostring
from scrapli_netconf.driver import NetconfDriver

config = """
policy-options {
    policy-statement test {
        term 00 {
            then accept;
        }
        term 01 {
            then accept;
        }
        term 02 {
            then accept;
        }
        term 03 {
            then accept;
        }
        term 04 {
            then accept;
        }
        term 05 {
            then accept;
        }
        term 06 {
            then accept;
        }
        term 07 {
            then accept;
        }
        term 08 {
            then accept;
        }
        term 09 {
            then accept;
        }
        term 10 {
            then accept;
        }
        term 11 {
            then accept;
        }
        term 12 {
            then accept;
        }
    }
}
"""

conn = NetconfDriver(
    "192.168.1.19",
    strip_namespaces=True,
    auth_username="root",
    auth_password="Password1!",
    auth_strict_key=False,
)

xml = Element("config-text")

xml.append(Element("configuration-text"))

xml[0].text = config

conn.open()
conn.lock("candidate")
conn.edit_config(tostring(xml).decode(), target="candidate")
conn.discard()
conn.unlock("candidate")
conn.close()
```

2. What you're connecting to (vendor, platform, version)
```
root@vmx09> show version 
Hostname: vmx09
Model: vmx
Junos: 20.4R1.12
```

3. Anything else relevant
Changing the configuration with one less term like below will resolve the issue.
```python
config = """
policy-options {
    policy-statement test {
        term 00 {
            then accept;
        }
        term 01 {
            then accept;
        }
        term 02 {
            then accept;
        }
        term 03 {
            then accept;
        }
        term 04 {
            then accept;
        }
        term 05 {
            then accept;
        }
        term 06 {
            then accept;
        }
        term 07 {
            then accept;
        }
        term 08 {
            then accept;
        }
        term 09 {
            then accept;
        }
        term 10 {
            then accept;
        }
        term 11 {
            then accept;
        }
    }
}
"""
```

**Expected behavior**
A Netconf Response to be returned by `edit_config` regardless of the length of `config` passed to it.

**Stack Trace**
```
Traceback (most recent call last):
  File "/Users/test/Documents/code/test/test.py", line 66, in <module>
    conn.edit_config(tostring(xml).decode(), target="candidate")
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli_netconf/driver/sync_driver.py", line 188, in edit_config
    raw_response = self.channel.send_input_netconf(response.channel_input)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli_netconf/channel/sync_channel.py", line 316, in send_input_netconf
    buf, _ = super().send_input(
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/decorators.py", line 284, in decorate
    return wrapped_func(*args, **kwargs)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/channel/sync_channel.py", line 436, in send_input
    _buf_until_input = self._read_until_input(channel_input=bytes_channel_input)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli_netconf/channel/sync_channel.py", line 291, in _read_until_input
    output += self.read()
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/channel/sync_channel.py", line 68, in read
    buf = self.transport.read()
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/decorators.py", line 103, in decorate
    return wrapped_func(*args, **kwargs)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/transport/plugins/system/transport.py", line 126, in read
    buf = self.session.read(65535)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/transport/plugins/system/ptyprocess.py", line 424, in read
    s = self.fileobj.read1(size)
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/decorators.py", line 190, in _signal_raise_exception
    self._handle_timeout()
  File "/Users/test/Documents/code/test/.venv/lib/python3.9/site-packages/scrapli/decorators.py", line 147, in _handle_timeout
    raise ScrapliTimeout(self.message)
scrapli.exceptions.ScrapliTimeout: timed out reading from transport
```

**OS (please complete the following information):**
 - OS: macOS version 11.2.3
 - Python version 3.9.2
 - scrapli version 2021.1.30
 - scrapli_netconf version 2021.1.30

**Comment by carlmontanari** [comment]
Hey @mobious01 thanks for opening this!

I believe this is the same issue as another user on slack ran into... you can try to use `ssh2` or `paramiko` transports to see if this issue is resolved? To change transport you just need to add it to the constructor as shown below:

```
conn = NetconfDriver(
    "192.168.1.19",
    strip_namespaces=True,
    auth_username="root",
    auth_password="Password1!",
    auth_strict_key=False,
    transport="ssh2"
)
```

*If* that does indeed fix it, could you do me a favor and snag logs and the channel log (from using ssh2/paramiko AND system transport if possible) and attach them here? I'll use this info ++ what I got from the user on slack to try to fix the system transport.

The tl;dr for why this is happening (assuming same issue) is that for some reason that I dont know yet when using the system transport we seem to not be able to read the entire input we sent to the device back off of the channel. We always(* not entirely true, but for this case w/ system transport we can assume it is true) write the input we want to send, then read it back off the channel to make sure it is all good before sending the return character after it. So we basically are timing out trying to make sure what we sent did in fact get sent. When we fail to read the entire chunk we sent to the device we of course ultimately hit the timeout.

I should be able to try to reproduce this on vSRX this weekend and also on NX-OS (as the slack users issue was on NX-OS) and hopefully get a fix for it.

Thanks again for opening this!

Carl

**Comment by mobious01** [comment]
I switched to `ssh2` and it worked. Just had to change `port` to `22`. I have attached the Channel and Regular logs. Would you like me to try `paramiko` as well and provide the logs for it?

[scrapli_channel.log](https://github.com/scrapli/scrapli_netconf/files/6232334/scrapli_channel.log)
[scrapli.log](https://github.com/scrapli/scrapli_netconf/files/6232335/scrapli.log)

**Comment by carlmontanari** [comment]
Awesome thank you! Nah, no need to test with paramiko as well -- I think this pretty much confirms my suspicions.

Hopefully ssh2 can hold ya over for a while -- I will try to find and fix this this weekend if possible!

Thanks for the quick turnaround on logs, I'll keep ya posted!

Carl

**Comment by carlmontanari** [comment]
I *think* (hope!) that this should be fixed for system transport in develop branch now!

TL;DR is that we were hitting 1024 chars of input (we remove whitespace and stuff so the big payload you had was actually way less chars w/out that) and ssh would hang then die. You can recreate it manually if you want to test out the fun yourself too haha. One possible fix for this would be to spawn a session then run `stty -icanon` to enable (or disable?) canonical mode so we can send big commands. This doesn't really work for scrapli though, so instead it seems that forcing a pseudo terminal w/ -tt in the ssh command fixes this for us. 

If you could test out develop branch and let us know if that sorts it out that would be much appreciated!

Carl

**Comment by mobious01** [comment]
I used the develop branch using the `pip` command below and it resolved the issue.
```
pip install -e git+https://github.com/scrapli/scrapli_netconf.git@develop#egg=scrapli_netconf
```

**Comment by carlmontanari** [comment]
Awesome! Thanks for confirming. I’ll go ahead and close this now. Thanks again for opening this and the help, feel free to re-open if it turns out things aren’t working as expected!

carl
